__app__/crop/ingress_hyper.py

```diff
@@ -0,0 +1,211 @@
+"""
+Python module to import data using the Hyper.ag API
+"""
+
+from collections.abc import Iterable
+import logging
+import time
+from datetime import datetime, timedelta
+import requests
+import pandas as pd
+
+from sqlalchemy import and_
+from sqlalchemy.exc import ProgrammingError
+
+from __app__.crop.db import connect_db, session_open, session_close
+from __app__.crop.structure import (
+    TypeClass,
+    SensorClass,
+    ReadingsAranetTRHClass,
+)
+from __app__.crop.utils import query_result_to_array
+from __app__.crop.constants import CONST_CROP_HYPER_APIKEY, CONST_ARANET_TRH_SENSOR_TYPE
+
+from __app__.crop.ingress import log_upload_event
+from __app__.crop.sensors import get_aranet_trh_sensor_data
+
+CONST_CHECK_URL_PATH = "https://zcf.hyper.ag/api/sites/0/analytics/v3/device_metrics"
+
+
+def get_api_sensor_data(api_key, dt_from, dt_to):
+    """
+    Makes a request to download sensor data for a specified period of time.
+    Note that unlike the equivalent Zensie function, this gets data for _all_ sensors,
+    and returns it as a dict, keyed by Aranet id.
+
+    Arguments:
+        api_key: api key for authentication
+        dt_from: date range from
+        dt_to: date range to
+    Return:
+        success: whether data request was succesful
+        error: error message
+        data_df_dict: dictionary of {aranet_pro_id (str): DataFrame of sensor_data}
+    """
+
+    success = True
+    error = ""
+    data_df_dict = {}
+
+    headers = {
+        "Content-Type": "application/json",
+        "Authorization": "Bearer " + api_key,
+    }
+
+    dt_from_iso = dt_from.strftime("%Y-%m-%dT%H:%M:%S") + "Z"
+    dt_to_iso = dt_to.strftime("%Y-%m-%dT%H:%M:%S") + "Z"
+
+    url = CONST_CHECK_URL_PATH
+
+    params = {
+        "start_time": dt_from_iso,
+        "end_time": dt_to_iso,
+        "metrics": "aranet_temperature,aranet_humidity",
```

Comment:
Another thing I just realised is that we should probably use a "fully qualified name" for these metrics, specifically:

- `aranet_ambient_temperature`
- `aranet_relative_humidity`

This is going to make it easier, for example, to add new kind of temperature sensors in the future like `water_temperature`. Currently `aranet_temperature` will lookup any kind of "temperature" sensor.

Reply:
Good point-  doing this now.